node-red-contrib-c8y-client/src/nodes/notification/notification.js

```diff
@@ -0,0 +1,400 @@
+const { c8yClientLib, InventoryService } = require("@c8y/client");
+const { getCredentials } = require("../c8y-utils/c8y-utils");
+const Websocket = require("ws");
+const uuid = require("uuid");
+
+module.exports = function (RED) {
+  function notificationNode(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+    node.config = config;
+    node.c8yconfig = RED.nodes.getNode(node.config.c8yconfig);
+    node.subscriber = node.config.subscriber;
+    node.subscription = node.config.subscription;
+    node.nonPersistent = node.config.nonPersistent;
+    node.typeFilter = node.config.typeFilter;
+    node.fragmentsToCopy = node.config.fragmentsToCopy;
+    node.socket = undefined;
+    node.clientId = "nodeRed" + uuid.v4().replace(/-/g, "");
+    node.reconnectTimeout = 10000;
+    node.pingInterval = 180000;
+    node.heartBeatReference = undefined;
+    node.refreshTokenIntervalReference = undefined;
+    node.reconnectCount = 0;
+    node.expiresInMinutes = 10;
+    node.refreshTokenInterval = 600000;
+
+    getCredentials(RED, node);
+    console.log();
+    try {
+      // Get properties
+      node.deviceIds = RED.util.evaluateNodeProperty(
+        node.config.deviceIds,
+        node.config.deviceIdsType,
+        node,
+        undefined
+      );
+      node.deviceIds = node.deviceIds.split(",").filter((s) => s.trim());
```

Comment:
What are you trying to do with `.filter((s) => s.trim())`?
If you want to remove spaces `.map((s) => s.trim())` might be what you are looking for..

Reply:
Done